apps/smart-contracts/token/test/MiniSales.test.ts

```diff
@@ -435,4 +455,180 @@ describe('=> MiniSales', () => {
+    let unblockedUser1: SignerWithAddress
+    let allowedPurchaser: SignerWithAddress
+
+    const getSignersAndMockTokenForEndToEndStack = async (): Promise<void> => {
+      await setupSignersAndMockTokens()
+      ;[allowedSourceUser, allowedDestinationUser, blockedUser, unblockedUser1, allowedPurchaser] =
+        await ethers.getSigners()
+    }
+
+    const setupEndToEndStack = async (): Promise<void> => {
+      await getSignersAndMockTokenForEndToEndStack()
+      // Setup PPO
+      ppo = await ppoFixture('prePO Token', 'PPO', owner.address)
+      await ppo.connect(owner).acceptOwnership()
+      // Setup MiniSales with PPO as saleToken
+      miniSales = await miniSalesFixture(
+        ppo.address,
+        paymentToken.address,
+        saleTokenDecimals,
+        owner.address
+      )
+      await miniSales.connect(owner).acceptOwnership()
+      // Setup purchase allowlist
+      purchaseAllowlist = await accountListFixture(owner.address)
+      await purchaseAllowlist.connect(owner).acceptOwnership()
+      await purchaseAllowlist
+        .connect(owner)
+        .set([owner.address, allowedPurchaser.address], [true, true])
+      // Setup allow list purchase hook
+      allowlistPurchaseHook = await allowlistPurchaseHookFixture(owner.address)
+      await allowlistPurchaseHook.connect(owner).acceptOwnership()
+      await allowlistPurchaseHook.connect(owner).setAllowlist(purchaseAllowlist.address)
+      // Setup Source allow list
+      sourceAllowlist = await accountListFixture(owner.address)
+      await sourceAllowlist.connect(owner).acceptOwnership()
+      await sourceAllowlist
+        .connect(owner)
+        .set(
+          [ZERO_ADDRESS, owner.address, miniSales.address, allowedSourceUser.address],
+          [true, true, true, true]
+        )
+      // Setup Destination allow list
+      destinationAllowlist = await accountListFixture(owner.address)
+      await destinationAllowlist.connect(owner).acceptOwnership()
+      await destinationAllowlist
+        .connect(owner)
+        .set([owner.address, allowedDestinationUser.address], [true, true])
+      // Setup Blocklist
+      blocklist = await accountListFixture(owner.address)
+      await blocklist.connect(owner).acceptOwnership()
+      await blocklist.connect(owner).set([blockedUser.address], [true])
+      // Setup RestrictedTransferHook
+      restrictedTransferHook = await restrictedTransferHookFixture(owner.address)
+      await restrictedTransferHook.connect(owner).acceptOwnership()
+      await restrictedTransferHook.connect(owner).setSourceAllowlist(sourceAllowlist.address)
+      await restrictedTransferHook
+        .connect(owner)
+        .setDestinationAllowlist(destinationAllowlist.address)
+      await restrictedTransferHook.connect(owner).setBlocklist(blocklist.address)
+      // Connect PPO with TransferHook
+      await ppo.connect(owner).setTransferHook(restrictedTransferHook.address)
+    }
+
+    beforeEach(async () => {
```

Comment:
```suggestion
    beforeEach(async () => {
```
I would put this section and the section in PR 301 under their own parent `describe` block to make it clear that these are tests for unpermissioned sales. Describe nesting becomes more important when we have so many different access controls we need to test together.